### DISCUSSION TITLE
Any way to load the JS SDK authStore directly from the Authorization header?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#3597] [Q&A] [answered]
I want to write a secure API endpoint in expressjs instead of pb_hooks. 

I need to verify the user's authorization before I perform some pocketbase operations on their behalf. In other words, the user record is not authorized to do these operations, it must be done by a server-side admin role. 

So I need to do this:

1. Request comes in
2. Validate `Authorization` header and get user's ID
3. Use a `PocketBase` client already authenticated as admin to insert various records and perform various privileged operations, setting ownership to the incoming User ID

I see this can be done by exporting cookies, but when I do `pb.send(`/foo`, {method: 'POST'})`, I see the `Authorization` header is already sent.

From the server, can I make a simple REST call to https://pocketbase.io/docs/api-records/#auth-refresh, or is there a better way to do this?

**Comment by ganigeorgiev** [comment, marked as answer]
The auth refresh call can be used to verify a token via the web API. We don't have other mechanism at the moment.

The user state can be loaded either by calling `pb.authStore.save(token)` OR send a one-off `authRefresh()` request with a manually set `Authorization` header. To prevent mixing the admin and user `pb.authStore` states you can create a separate JS SDK instance for it.

```js
const adminClient = new PocketBase("...");

...

const userClient = new PocketBase(adminClient.baseUrl);

userClient.authStore.save(token)
// or send the token as explicit header and on success it will
// populate automatically the store with the latest data
userClient.collection("users").authRefresh({ headers: { Authorization: token } })
```

**Reply by ganigeorgiev** [reply]
_btw. I forgot to mention that if you use Deno, LocalStorage is implemented on the server-side so you may have to instantiate the stores with different key:_

```js
import PocketBase, { LocalAuthStore } from "pocketbase"

...

const userClient = new PocketBase(adminClient.baseUrl, new LocalAuthStore("pb_client_store"))
```
